compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -100,6 +100,23 @@ public STNode parse() {
         return parseCompUnit();
     }
 
+    /**
+     * Completely parses a given input a statement.
+     *
+     * @return Parsed node
+     */
+    public STNode parseAsStatement() {
+        STNodeList stmtNodeList = (STNodeList) parseAsStatements();
+        STNode stmt = stmtNodeList.get(0);
+
+        for (int i = 1; i < stmtNodeList.size(); i++) {
+            stmt = SyntaxErrors.cloneWithTrailingInvalidNodeMinutiae(stmt, stmtNodeList.get(i),
+            DiagnosticErrorCode.ERROR_SYNTAX_ERROR);
```

Comment:
1. Here used `parseAsStatements()` and then invalidate the extra statements. The reason for using it instead of `parseStatement()` is, there are validations we do on top of `parseStatement()` and we may not repeat them in another place?

2. Notice that here we have not given a specific error code, used the general error. Since this error is specific to this case and this does not go on compiler phases and log errors no point in declaring a new error? 

Although we declared a new one, we better not mix it with other errors. As we eventually give a doc or something to the users for diagnostic id vs description. It is odd to have NodeParser specific error mixed up among regular parser errors.

Reply:
Refactored to use `parseStatement()` inside with b45e13b